packages/block-editor/src/components/block-list/use-block-props/use-focus-first-element.js

```diff
@@ -80,9 +90,8 @@ export function useFocusFirstElement( { clientId, initialPosition } ) {
 				return;
 			}
 		}
-
 		placeCaretAtHorizontalEdge( target, isReverse );
-	}, [ initialPosition, clientId ] );
```

Comment:
I'm not sure we need to add another `prevMode` const to this effect. The thing I would like to understand is why does `initialPosition` change when the mode is switched...

Reply:
It looks like [`initialPosition` gets changed to `undefined` when zoom-out mode is activated](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/components/block-list/block.js#L623). `undefined` is the correct value while in zoom-out mode, and then it gets set back to 0 when edit mode is activated again. Maybe we should leave it as undefined after zoom out mode is exited?